src/System.Net.Requests/src/System/Net/WebClient.cs

```diff
@@ -0,0 +1,2127 @@
+        public byte[] UploadData(string address, string method, byte[] data) =>
+            UploadData(GetUri(address), method, data);
+
+        public byte[] UploadData(Uri address, string method, byte[] data)
+        {
+            ThrowIfNull(address, nameof(address));
+            ThrowIfNull(data, nameof(data));
+            if (method == null)
+            {
+                method = MapToDefaultMethod(address);
+            }
+
+            StartOperation();
+            try
+            {
+                WebRequest request;
+                return UploadDataInternal(address, method, data, out request);
+            }
+            finally
+            {
+                EndOperation();
+            }
+        }
+
+        private byte[] UploadDataInternal(Uri address, string method, byte[] data, out WebRequest request)
+        {
+            request = null;
+            try
+            {
+                _method = method;
+                _contentLength = data.Length;
+                request = _webRequest = GetWebRequest(GetUri(address));
+                return UploadBits(request, null, data, 0, null, null);
+            }
+            catch (Exception e) when (!(e is OutOfMemoryException))
+            {
+                AbortRequest(request);
+                if (e is WebException || e is SecurityException) throw;
+                throw new WebException(SR.net_webclient, e);
+            }
+        }
+
+        private void OpenFileInternal(
+            bool needsHeaderAndBoundary, string fileName,
+            ref FileStream fs, ref byte[] buffer, ref byte[] formHeaderBytes, ref byte[] boundaryBytes)
+        {
+            fileName = Path.GetFullPath(fileName);
+
+            WebHeaderCollection headers = Headers;
+            string contentType = headers[HttpKnownHeaderNames.ContentType];
+
+            if (contentType == null)
+            {
+                contentType = DefaultUploadFileContentType;
+            }
+            else if (contentType.StartsWith("multipart/", StringComparison.InvariantCultureIgnoreCase))
+            {
+                throw new WebException(SR.net_webclient_Multipart);
+            }
+
+            fs = new FileStream(fileName, FileMode.Open, FileAccess.Read);
+
+            int buffSize = DefaultCopyBufferLength;
+            _contentLength = -1;
+
+            if (string.Equals(_method, WebRequestMethods.Http.Post, StringComparison.InvariantCulture))
+            {
+                if (needsHeaderAndBoundary)
+                {
+                    string boundary = "---------------------" + DateTime.Now.Ticks.ToString("x", NumberFormatInfo.InvariantInfo);
+
+                    headers[HttpKnownHeaderNames.ContentType] = UploadFileContentType + "; boundary=" + boundary;
+
+                    string formHeader =
+                        "--" + boundary + "\r\n" +
+                        "Content-Disposition: form-data; name=\"file\"; filename=\"" + Path.GetFileName(fileName) + "\"\r\n" +
```

Comment:
I would leave the code as-is (same as Desktop). We can open an issue later if we think the filename encoding is wrong (or not per RFC, etc.).  We would also need to add test coverage to this as well.